Dockerfile

```diff
@@ -1,36 +1,117 @@
-FROM amd64/ubuntu:20.04
-
-# Set-up argument defaults
+ARG ALPINE_VERSION=3.18.3
 ARG JDK_VER=11
 
-# Install supporting packages
-RUN dpkg --add-architecture i386 \
-  && apt-get update  \
-  && DEBIAN_FRONTEND=noninteractive apt-get install -y \
-  bash \
-  make \
-  openjdk-${JDK_VER}-jre-headless \
-  wine32 \
-  && rm -rf /var/lib/apt/lists/*
+ARG BASE_IMAGE=ghcr.io/stephane-d/sgdk-m68k-gcc
+ARG BASE_IMAGE_VERSION=latest
 
-# Create sgdk unprivileged user
-RUN useradd -ms /bin/sh -d /sgdk sgdk
+# Stage Zero - just init image to download files in other stages
+FROM $BASE_IMAGE:$BASE_IMAGE_VERSION as m68k-files
+
+
+FROM alpine:$ALPINE_VERSION as jdk-base
+ARG JDK_VER
+RUN apk add --no-cache openjdk${JDK_VER}
+
+
+
+# Stage One - build all tools and libs for SGDK
+FROM jdk-base as build
+RUN apk add --no-cache build-base git
 
 # Set-up SGDK
-COPY . /sgdk
-ENV GDK=/sgdk
-ENV SGDK_DOCKER=y
+WORKDIR /sgdk
+COPY . .
```

Comment:
I'm reorganizing the steps so more of them can be cached.  When I change any one file in SGDK and rebuild, everything past this point must be rerun because the source folder changed and the cache of this step and those after it become stale.

Reply:
Done.